spec/ndx-ophys-devices.extensions.yaml

```diff
@@ -1,8 +1,216 @@
+  - name: peak_power_in_W
+    dtype: float
+    doc: Incident power of stimulation device (in Watts).
+    required: false
+  - name: peak_pulse_energy_in_J
+    dtype: float
+    doc: If device is pulsed light source, pulse energy (in Joules).
+    required: false
+  - name: intensity_in_W_per_m2
+    dtype: float
+    doc: Intensity of the excitation in W/m^2, if known.
+    required: false
+  - name: pulse_rate_in_Hz
+    dtype: float
+    doc: If device is pulsed light source, pulse rate (in Hz) used for stimulation.
+    required: false
+  - name: exposure_time_in_s
+    dtype: float
+    doc: Exposure time of the sample (in sec).
+    required: false
+- neurodata_type_def: Photodetector
+  neurodata_type_inc: Device
+  doc: Extends Device to hold metadata on the Photodetector.
+  attributes:
+  - name: model
+    dtype: text
+    doc: Model of photodetector device.
+    required: false
+  - name: detector_type
+    dtype: text
+    doc: 'Technology used to detect the light. Suggested values: CCD, Intensified
+      CCD, Analog video, PMT, Photodiode, CMOS, EBCCD, FTIR.'
+  - name: detected_wavelength_in_nm
+    dtype: float
+    doc: Wavelength detected by photodetector.
+  - name: gain
+    dtype: float
+    doc: Gain on the photodetector.
+- neurodata_type_def: DichroicMirror
+  neurodata_type_inc: Device
+  doc: Extends Device to hold a Dichroic Mirror.
+  attributes:
+  - name: cut_on_wavelength_in_nm
+    dtype: float
+    doc: Wavelength at which the mirror starts to transmit light more than reflect.
+    required: false
+  - name: cut_off_wavelength_in_nm
+    dtype: float
+    doc: Wavelength at which transmission shifts back to reflection,for mirrors with
+      complex transmission spectra.
+    required: false
+  - name: reflection_bandwidth_in_nm
+    dtype: float
+    doc: The range of wavelengths that are primarily reflected.The start and end wavelengths
+      needs to be specified.
+    required: false
+  - name: transmission_bandwidth_in_nm
+    dtype: float
+    doc: The range of wavelengths that are primarily transmitted.The start and end
+      wavelengths needs to be specified.
+    required: false
+  - name: angle_of_incidence_in_degrees
+    dtype: float
+    doc: Intended angle at which light strikes the mirror.
+    required: false
+  - name: model
+    dtype: text
+    doc: Model of the dichroic mirror.
+    required: false
+- neurodata_type_def: BandOpticalFilter
+  neurodata_type_inc: Device
+  doc: Extends Device to hold a Band Optical Filter (Bandpass or Bandstop).
+  attributes:
+  - name: center_wavelength_in_nm
+    dtype: float
+    doc: The midpoint of the band of wavelengths that the filter transmits or blocks.
+  - name: bandwidth_in_nm
+    dtype: float
+    doc: The width of the wavelength range that the filter transmits or blocks (full
+      width at half maximum).
+  - name: filter_type
+    dtype: text
+    doc: Type of filter (e.g., 'Bandpass', 'Bandstop').
+  - name: model
```

Comment:
`model` is starting to become prolific here

I'd recommend defining a `ndx-ophys-devices` specific base class that extends the core device with `model` (maybe call it `DeviceModel` or w/e else makes more sense to you) and then all the others here can just inherit from that